midiproc/main.c

```diff
@@ -55,6 +55,22 @@ static Mix_Music *music  = NULL;
 //
 
 //
+// Write an unsigned integer into a simple CHAR buffer.
+//
+static boolean WriteInt16(CHAR *out, size_t osize, unsigned int in)
```

Comment:
This is much clearer in a separate function, thanks. I'd add a note about the endianness here, though: you're explicitly writing out a big-endian here, because that's the fmt used on the pipe, right?